docker-compose.yml

```diff
@@ -1,7 +1,9 @@
 version: '3'
 services:
   wildbeast:
-    build: .
+    image: dougley/wildbeast
```

Comment:
Is using image over build better? Cause I mean build uses the local files which the user most likely just cloned, image will download a built image over the net which could be an issue for users with slow / limited network access, am I incorrect in this?

Reply:
Building from the working directory is better depending on the situation, if the host OS is Windows you might be better off downloading the official image due to it being built against a real Linux host. But in general building the image is better.

Reply:
It's a case of what has the least amount of faff involved.

**If we go for local builds:** This option forces us to choose between two bad options.
- If one is built and the other isn't, the install size approximately triples as one pulls the image and the other builds locally, resulting in a ridiculously large amount of overhead. At the same time the local build is made pointless because it takes a shorter time and less network transfer to just pull the Docker image.
- If both are built, the installation time in turn triples as both the `install` and `wildbeast` images need to perform a full build procedure (Pull the Node image, do all the node-gyp builds, npm install, the lot).

**If we go for image pulls:** This option does mean there is more network transfer for the end user, but that is paid back with a quicker installation procedure overall. This also means that tearing down the environment and rebuilding it takes a matter of seconds as opposed to several minutes.

In light of this evidence, I'd vote to use image pulls.

Reply:
Do you have evidence that building the images locally is more disk intensive? From my testing only layer caching takes extra disk space.

Reply:
Building images locally explicitly is not more disk intensive - mixing local and image is, as per this screenshot I sent you this morning.

![Overhead screenshot](https://i.linuswillner.me/Nm0SLQg.png)

However, the build time triples, as explained in my previous comment.

Reply:
Pushed a commit just now that should resolve this debate. Anything to add?